types/node/timers.d.ts

```diff
@@ -13,4 +13,54 @@ declare module "timers" {
         function __promisify__<T>(value: T): Promise<T>;
     }
     function clearImmediate(immediateId: NodeJS.Immediate): void;
+
+    global {
+        namespace NodeJS {
+            interface Global {
+                clearImmediate: (immediateId: Immediate) => void;
+                clearInterval: (intervalId: Timeout) => void;
+                clearTimeout: (timeoutId: Timeout) => void;
+
+                setImmediate: (callback: (...args: any[]) => void, ...args: any[]) => Immediate;
+                setInterval: (callback: (...args: any[]) => void, ms: number, ...args: any[]) => Timeout;
+                setTimeout: (callback: (...args: any[]) => void, ms: number, ...args: any[]) => Timeout;
+            }
+
+            interface RefCounted {
+                ref(): this;
+                unref(): this;
+            }
+
+            // compatibility with older typings
+            interface Timer extends RefCounted {
+                hasRef(): boolean;
+                refresh(): this;
+            }
+
+            interface Immediate extends RefCounted {
+                hasRef(): boolean;
+                _onImmediate: Function; // to distinguish it from the Timeout class
+            }
+
+            interface Timeout extends Timer {
+                hasRef(): boolean;
+                refresh(): this;
+            }
+        }
+    }
+}
+
+declare function setTimeout(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timeout;
+declare namespace setTimeout {
+    function __promisify__(ms: number): Promise<void>;
+    function __promisify__<T>(ms: number, value: T): Promise<T>;
+}
+declare function clearTimeout(timeoutId: NodeJS.Timeout): void;
+declare function setInterval(callback: (...args: any[]) => void, ms: number, ...args: any[]): NodeJS.Timeout;
+declare function clearInterval(intervalId: NodeJS.Timeout): void;
+declare function setImmediate(callback: (...args: any[]) => void, ...args: any[]): NodeJS.Immediate;
+declare namespace setImmediate {
+    function __promisify__(): Promise<void>;
+    function __promisify__<T>(value: T): Promise<T>;
 }
+declare function clearImmediate(immediateId: NodeJS.Immediate): void;
```

Comment:
This should be moved into the `global` block inside `module "timers"`.

Reply:
Done